array_test.go

```diff
@@ -666,7 +682,11 @@ func TestArray_IsEqualUnordered(t *testing.T) {
 		value.chain.assertFailed(t)
 		value.chain.clearFailed()
 
-		value.NotEqualUnordered(func() {})
```

Comment:
I guess the block:

```go
		value.NotEqualUnordered(func() {})
		value.chain.assertFailed(t)
		value.chain.clearFailed()
```

was lost accidentally, we should restore it.

Reply:
added this back